### PR TITLE
Concise error output

### LINKED ISSUE #2634: tools should be less verbose when errors occur
Currently tools like wasm-opt have a tendency to dump the entire program to stdout when they fail.

See: https://github.com/emscripten-core/emscripten/issues/10331

For normal users this is almost never what they want or useful.  Perhaps we should different modes?  Perhaps only print a summary/snippet by default?

### PR DESCRIPTION
**Pull request by kripken**
Don't print the entire module on an error. Instead, just print
the validation errors.

However, if the user passed `--print`, then do print it, as otherwise
nothing would get printed - the error would be before the pass
to print happens. And in general a user passing in a request
to print would expect a printed module anyhow.

fixes #2634

### CODE REVIEW

File: src/tools/wasm-opt.cpp [hunk 1]
```diff
@@ -242,8 +253,7 @@ int main(int argc, const char* argv[]) {
 
     if (options.passOptions.validate) {
       if (!WasmValidator().validate(wasm)) {
-        WasmPrinter::printModule(&wasm);
-        Fatal() << "error in validating input";
+        exitOnInvalidWasm("error in validating input");
```
Comment (sbc100) [inline src/tools/wasm-opt.cpp]: I think removing the `in` from these two error message would read better.

File: test/unit/test_errors.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import os
```
Comment (sbc100) [inline test/unit/test_errors.py]: add copyright header

Comment (kripken) [reply test/unit/test_errors.py]: I think we discussed this already - none of the files there have one, so I didn't add one either to be consistent. Did we decide they all need one? I don't remember.

Comment (sbc100) [reply test/unit/test_errors.py]: OK lets deal with that issue separately.

**Pull request merged**